### ISSUE TITLE
Modify how Poly mode parameter works

### ISSUE THREAD

**Issue by mkruselj** [surge-synthesizer/surge#679]
![image](https://user-images.githubusercontent.com/2393720/53491450-be140800-3a96-11e9-893a-76319c45e099.png)

First off, it should probably be called Poly Mode, or Play Mode, rather than Polymode (similar to Patchbrowser thing, haha).

Next, there are multiple monophonic modes for no good reason. ST (single trigger) should actually be a property of envelopes, rather than conflating it to play mode. and FP only relates to portamento (fingered portamento, or portamento only happening when you play legato), so it should be closer to the Portamento slider, rather than here... So ideally this parameter should just have three values: poly, mono, latch, with single trigger actually being calculated per envelope (needs a switch for filter and amp EG added, should just be called Retrigger), and FP should be a Legato button somewhere close to Portamento slider.

Yes, a GUI layout redesign is absolutely necessary to sort this one out. But it will result in a more sensible UX, IMHO.

**Comment by mkruselj** [comment]
Also a simpler variant would be not having ST a property of envelopes, and then we have Poly/Mono/Legato/Latch (which is far more clearer to just about everyone). And FP becomes Legato button next to Portamento slider. 1.9 material, easy :slight_smile:

**Comment by Andreya-Autumn** [comment]
I wanna brainbarf a bit ok?. :) I thought a bunch about what would be a really good way to present this, as usual with the intention of keeping things intuitive/simple, but also flexible and powerful. Here’s a suggestion. 

Make an option tree with these options: MONO/POLY/PARA/LATCH

In MONO you have:
Retrig <-> Don’t
If retrig: from Zero <->Current
+Current Priority Options (minus “Legacy”?)

In POLY you have:
Voice stack <-> Voice Reuse
Max Voice Count

In PARA you have:
Exactly the same options as mono + max voice count. Set voice-count to 2 for #6333 for example. :)
Voices stack on the same key according to priority, otherwise spread out RR-style across held keys. 

In LATCH:
No options needed right?

**Comment by Andreya-Autumn** [comment]
Then Portamento should get the options:
Always Glide
Glide overlapping sound only
Glide overlapping MIDI note only
+the ones already in its RMB context.


And about retriggering:

Give every retrigable EG/modulator an (RMB?) option to negatively override retrig
If this defaults to off, this maintains the intuitive/simple workflow where everything obeys the Retrig options under mono (and always retrigs from current in reuse-poly, but still makes it possible to let  certain modulators only trigger once.



What do you think of this? :) I expect @mkruselj is gonna go to town on the naming schemes, but otherwise?

**Comment by mkruselj** [comment]
I think it's too complicated with those portamento modes and retrig overrides,  sorry. PARA doesn't really "do" paraphonic the way you described it, since you still get multiple voices happening and there's no actual sharing of filter and amp between the voices. 

The plan for XT2 is to add another play mode option called Release, which triggers the scene on note off. Glide would get an additional switch which would replace the current FP play modes (or rather,  would work in conjunction with the two "new" monophonic modes to result in the old mono FP modes).

So final list of play modes would be: Poly,  Mono,  Legato (ex single trigger) , Latch, Release.

**Comment by Andreya-Autumn** [comment]
Well, the portamento modes is not something I'd be particularly attached to. Just dawned on me today that both behaviors could be useful.

Should have probably clarified that the PARA options would be relevant in the case that we decide there should be an actual paraphonic option, with the filters and other things shared. As in the request I linked. If that happens, I figure it'd need the same behavioral options as mono mode + importantly a max voice count param. That's what I was trying to say really. Whether that should be done can be discussed of course. I'd be for it, but we'll see. ;) 

But I disagree about the envelope retrig override. I think it's a sensible way to keep a simple and usable default behavior while still allowing some deeper patch design options. 
I've found myself wishing I could do this in the new poly mode a lot, retrigging the AEG and FEG and maybe some but not all LFO's for example. I think others could want it too!

**Comment by Lo-lo78** [comment]
@Andreya-Autumn 
You reminded me of the polyphonic portamento of Roland synths.
In the Roland Juno 106 there was Poly 1 / Rotate and Poly 2 / Reset.
Poly 1 behaves like Surge. Portamento is always there for all polyphony and the envelope is never reset. The notes overlap.
While poly 2 is very interesting. If you play a 3-note chord on a 6-note polyphony, the next 3-note chord resets the previous 3 notes.
Basically with poly 2 you can make chords with the portamento without the release overlapping the previous notes.
Wow! If there was poly 2 of Roland Surge synths it will be my favorite virtual synth.
Not a bad idea!

**Comment by mkruselj** [comment]
We have those modes already, added them in the latest version of Surge. Right click the Play Mode and you will have an option to reuse voices in Poly mode.

**Comment by Lo-lo78** [comment]
@mkruselj 
I've tried everything about portamento with polyphony but there is no Roland poly 2 functionality.
The release overlaps with subsequent notes. The poly 2 of the Juno 106 Roland blocks the release for new notes.
Obvious. If you play 3 notes first and then 2 notes one will be free to play with its release.
On the other hand, if you play 2 notes with a long release, the next two notes disable the release. The result is a polyphonic portamento like the monophonic one.

**Comment by mkruselj** [comment]
I did say "it's not the same as Poly 2 mode"! It's better to use the aforementioned TAL synths if you want that feature.
